solo-chains/runtime/dancelight/src/lib.rs

```diff
@@ -2045,6 +2045,9 @@ impl pallet_registrar::Config for Runtime {
     type DepositAmount = DepositAmount;
     type RegistrarHooks = DancelightRegistrarHooks;
     type RuntimeHoldReason = RuntimeHoldReason;
+    #[cfg(feature = "runtime-benchmarks")]
+    type InnerRegistrar = ();
+    #[cfg(not(feature = "runtime-benchmarks"))]
```

Comment:
I believe the weight of `T::InnerRegistrar::register` is not accounted for if we do it like this, so we need to fix the benchmark in some other way

Reply:
yes indeed

Reply:
however, why would we need to change this? why does pallet-registrar influence whatsoever on the benchmarks of data preservers? I get that you need a chain registered for this to work, but we can probably set it up as a benchmark helper in pallet-data-preservers

Reply:
Either we benchmark the pallet including the hooks, or we return the weight from `T::InnerRegistrar::register` and benchmark that separately.

I find it simpler to benchmark the pallet including the hooks, this way we don't need to do manual weight tracking, and also the benchmark is more similar to production so more accurate in my opinion.

Reply:
What do you mean? Dancebox always has `InnerRegistrar = ()`. But if we touch the benchmark it can break so we should make sure it still works after any changes we end up doing.

Reply:
Cool thanks for clarifying that !

Reply:
Resolved in https://github.com/moondance-labs/tanssi/pull/783/commits/955f90f481d86d5cc9c6447178e443ff4c2d94b5, [075b3f7](https://github.com/moondance-labs/tanssi/pull/783/commits/075b3f7470afed009295a3d8e5b99da334f15e9d), and https://github.com/moondance-labs/tanssi/pull/783/commits/a010dc878707db1a4670266ce9823cb6783ca76b